docs/book/v3/features/container/factories.md

```diff
@@ -510,3 +526,42 @@ When creating the `PhpRenderer` instance, it will inject it with a
 instantiated directly). It injects the helper plugin manager with custom url and
 serverurl helpers, `Mezzio\LaminasView\UrlHelper` and
 `Mezzio\LaminasView\ServerUrlHelper`, respectively.
+
+#### FilterUsingXForwardedHeadersFactory
+
+Diactoros also ships with a factory for generating a customized `Laminas\Diactoros\ServerRequestFilter\FilterUsingXForwardedHeaders` instance via the `Laminas\Diactoros\ServerRequestFilter\FilterUsingXForwardedHeadersFactory` class.
+This factory looks for the following configuration in order to generate an instance:
+
+```php
+$config = [
+    'laminas-diactoros' => [
+        'server-request-filter' => [
+            'x-forwarded-headers' => [
+                'trusted-proxies' => string|string[],
```

Comment:
argh, `string|string[]` survived? @weierophinney let's please keep signatures small in future - if a union type is not **necessary**, it shouldn't be added.

Too late for that here.

Reply:
No - this is specific to the factory implementation in this package.

I can require `list<string>` here, though.

Reply:
Let's simplify to the bone then - allowing multiple types doesn't bring us much

Reply:
Done.